common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -416,7 +416,10 @@ protected void finalize() throws Throwable {
         // to scavenge those that can be reused. this permits us to incur minimal thread synchronisation whilst
         // still recycling all items.
         final Recycler<T> parent;
-        final Thread thread;
+
+        // We store the Thread in a WeakReference as otherwise we may be the only ones that still hold a strong
+        // Reference to the Thread itself after it died.
+        final WeakReference<Thread> threadRef;
```

Comment:
should this be in the same pull request?

Reply:
I did it in the same as I noticed it here because the Thread was never GCed. Maybe I should do it in a separate PR tho..

Reply:
@johnou https://github.com/netty/netty/pull/7499 ... Even added a test-case :)